lib/active_model_serializers/adapter/json_api/deserialization.rb

```diff
@@ -182,23 +182,29 @@ def parse_relationship(assoc_name, assoc_data, options)
           prefix_key = field_key(assoc_name, options).to_s.singularize
           hash =
             if assoc_data.is_a?(Array)
-              { "#{prefix_key}_ids".to_sym => assoc_data.map { |ri| ri['id'] } }
+              { "#{prefix_key}_ids".to_sym => assoc_data.map { |ri| ri[:id] } }
```

Comment:
because `KeyTransform#underscore` (and every other method in `KeyTransform`) transforms all keys to symbols, so string key access does not work here